gateway/upstreamer/push/upstreamer.go

```diff
@@ -72,32 +72,33 @@ func (c *Upstreamer) Upstream(req *http.Request) (string, float64) {
 	epi1 := c.apis[identity][n1]
 	epi2 := c.apis[identity][n2]
 
-	var address string
-	var load float64
-
 	epi1.RLock()
 	epi2.RLock()
+	defer epi1.RUnlock()
```

Comment:
defer can be expensive, which is why there was the variables. please use the previous design